### ISSUE TITLE
Object of type 'generator' has no len() 

### ISSUE THREAD

**Issue by TaXxER** [automl/auto-sklearn#234]
Auto-sklearn was running fine for the first couple of hours and steadily improving in accuracy of the best model. There were however some 'You are already timing task:index_run[x]' messages in the console, with [x] some integer. However, when I checked progress again several hours later, I noticed the following console output:

```
You are already timing task: index_run2
You are already timing task: index_run3
You are already timing task: index_run3
You are already timing task: index_run3
You are already timing task: index_run3
You are already timing task: index_run4
You are already timing task: index_run4
You are already timing task: index_run4
You are already timing task: index_run4
You are already timing task: index_run4
[ERROR] [2017-02-16 01:31:19,335:AutoMLSMBO(1)::BRADY] object of type 'generator' has no len()
[ERROR] [2017-02-16 01:31:19,335:AutoMLSMBO(1)::BRADY] Error in getting next configurations with SMAC. Using random configuration!
You are already timing task: index_run4
[ERROR] [2017-02-16 01:53:35,979:AutoMLSMBO(1)::BRADY] object of type 'generator' has no len()
[ERROR] [2017-02-16 01:53:35,980:AutoMLSMBO(1)::BRADY] Error in getting next configurations with SMAC. Using random configuration!
You are already timing task: index_run4
[ERROR] [2017-02-16 01:55:33,747:AutoMLSMBO(1)::BRADY] object of type 'generator' has no len()
[ERROR] [2017-02-16 01:55:33,748:AutoMLSMBO(1)::BRADY] Error in getting next configurations with SMAC. Using random configuration!
[ERROR] [2017-02-16 02:19:34,993:AutoMLSMBO(1)::BRADY] object of type 'generator' has no len()
[ERROR] [2017-02-16 02:19:34,994:AutoMLSMBO(1)::BRADY] Error in getting next configurations with SMAC. Using random configuration!
You are already timing task: index_run4
[ERROR] [2017-02-16 02:19:50,667:AutoMLSMBO(1)::BRADY] object of type 'generator' has no len()
[ERROR] [2017-02-16 02:19:50,668:AutoMLSMBO(1)::BRADY] Error in getting next configurations with SMAC. Using random configuration!
[ERROR] [2017-02-16 02:43:52,014:AutoMLSMBO(1)::BRADY] object of type 'generator' has no len()
[ERROR] [2017-02-16 02:43:52,015:AutoMLSMBO(1)::BRADY] Error in getting next configurations with SMAC. Using random configuration!
[ERROR] [2017-02-16 02:50:36,607:AutoMLSMBO(1)::BRADY] object of type 'generator' has no len()
[ERROR] [2017-02-16 02:50:36,608:AutoMLSMBO(1)::BRADY] Error in getting next configurations with SMAC. Using random configuration!
You are already timing task: index_run4
[ERROR] [2017-02-16 03:14:37,896:AutoMLSMBO(1)::BRADY] object of type 'generator' has no len()
[ERROR] [2017-02-16 03:14:37,897:AutoMLSMBO(1)::BRADY] Error in getting next configurations with SMAC. Using random configuration!
[ERROR] [2017-02-16 03:38:39,214:AutoMLSMBO(1)::BRADY] object of type 'generator' has no len()
[ERROR] [2017-02-16 03:38:39,216:AutoMLSMBO(1)::BRADY] Error in getting next configurations with SMAC. Using random configuration!
You are already timing task: index_run4
[ERROR] [2017-02-16 03:39:30,969:AutoMLSMBO(1)::BRADY] object of type 'generator' has no len()
[ERROR] [2017-02-16 03:39:30,970:AutoMLSMBO(1)::BRADY] Error in getting next configurations with SMAC. Using random configuration!
You are already timing task: index_run4
[ERROR] [2017-02-16 03:45:39,179:AutoMLSMBO(1)::BRADY] object of type 'generator' has no len()
[ERROR] [2017-02-16 03:45:39,180:AutoMLSMBO(1)::BRADY] Error in getting next configurations with SMAC. Using random configuration!
You are already timing task: index_run4
[ERROR] [2017-02-16 03:51:26,079:AutoMLSMBO(1)::BRADY] object of type 'generator' has no len()
[ERROR] [2017-02-16 03:51:26,079:AutoMLSMBO(1)::BRADY] Error in getting next configurations with SMAC. Using random configuration!
[ERROR] [2017-02-16 03:52:06,914:AutoMLSMBO(1)::BRADY] object of type 'generator' has no len()
[ERROR] [2017-02-16 03:52:06,915:AutoMLSMBO(1)::BRADY] Error in getting next configurations with SMAC. Using random configuration!
You are already timing task: index_run4
```

Then I checked the AutoML.log file in the tmp folder of  the run, trying to find some more information on what is going on. Here I found this at the time of the first of those errors being thrown:

```
[DEBUG] [2017-02-16 01:31:18,352:AutoMLSMBO(1)::BRADY] Return: Status: 2, cost: 1.000000, time. 1440.106536, additional: Timeout
[DEBUG] [2017-02-16 01:31:18,353:intensifier] Incumbent (0.1280) is better than challenger (1.0000) on 1 runs.
[INFO] [2017-02-16 01:31:18,354:intensifier] Updated estimated performance of incumbent on 1 runs: 0.1280
[INFO] [2017-02-16 01:31:18,425:AutoMLSMBO(1)::BRADY] Using 35 training points for SMAC.
[ERROR] [2017-02-16 01:31:19,335:AutoMLSMBO(1)::BRADY] object of type 'generator' has no len()
[ERROR] [2017-02-16 01:31:19,335:AutoMLSMBO(1)::BRADY] Error in getting next configurations with SMAC. Using random configuration!
```

So, it seems that this error is preventing my SMAC from falling back from an optimization through the search space of models/hyperparameters/preprocessings methods to a random search over this search space. This is confirmed by the fact that no single model improvement has been found between the first of those errors (01:31:19) until the last of those errors (03:52:06) after which I killed it. Although I'm not sure, it seems like the problem first occurs after it finished the initial configuration.

Unfortunately, no stack trace or line number of where this error occurred is provided in the output and log file, leaving me with no pointers for debugging.

I am running it on Python 3.5 on a Linux machine.

**Comment by mfeurer** [comment]
This issue comes from an incompatibility of the smac and the configspace version. Could you try again with `ConfigSpace==0.2.3`?

**Comment by TaXxER** [comment]
Thanks for responding so quickly. I installed that specific version of ConfigSpace and am currently running it. Please note that this version is not enforced in the requirement text file.

`https://raw.githubusercontent.com/automl/auto-sklearn/master/requirements.txt`

The SMAC version however is prescribed there. Therefore, people who will install from scratch now are likely to end up with the latest ConfigSpace version which is not compatible with SMAC.

**Comment by TaXxER** [comment]
It is currently running 11 hours without problems with ConfigSpace==0.2.3, so it seems like it is safe to assume that it runs problem free now.

**Comment by mfeurer** [comment]
This is fixed with the latest bugfix release 0.1.3.

**Comment by webanalytics** [comment]
I installed the most recent version in two different machines and got the same error message:
[ERROR] [2017-09-27 17:24:17,629:AutoMLSMBO(1)::boston] fit() got an unexpected keyword argument 'rng'
[ERROR] [2017-09-27 17:24:17,629:AutoMLSMBO(1)::boston] Error in getting next configurations with SMAC. Using random configuration!
---
I'm using python3.
Any suggestion on how to fix this problem?
Thanks.
